streaming/src/test/scala/org/apache/spark/streaming/ui/StreamingJobProgressListenerSuite.scala

```diff
@@ -116,4 +171,73 @@ class StreamingJobProgressListenerSuite extends TestSuiteBase with Matchers {
     listener.retainedCompletedBatches.size should be (limit)
     listener.numTotalCompletedBatches should be(limit + 10)
   }
```

Comment:
I meant that all the data structures are cleaned up based on the retained batches limit. I just want to ensure that there are no memory leaks.